src/Types.hs

```diff
@@ -5,25 +5,45 @@ module Types where
 
 import           Control.Monad.IO.Class (MonadIO, liftIO)
 import           Control.Monad.Reader (MonadIO, MonadReader, ReaderT)
+import           Data.Aeson (FromJSON(parseJSON), Value(..), object, (.=), (.:), (.:?))
 import           Data.Configurator.Types
-import           Data.Aeson (FromJSON(parseJSON), Value(..), object, (.=), (.:))
 import           Web.Scotty.Trans (ActionT)
 import qualified Data.Text.Lazy as L
 
+data HayooResult = HayooResult { fname :: String, ftype :: Maybe String, flocation :: String } deriving (Show)
```

Comment:
Are you looking for style review too? If so, this feels a bit long, and the parens in a single derivation are redundant.

```hs
data HayooResult = HayooResult
    { fname :: ...
    , ftype :: ...
    , flocation :: ...
    }
    deriving Show
```